src/Pool.sol

```diff
@@ -755,50 +657,46 @@ abstract contract Pool {
         }
         sender.startBalance -= (_currTimestamp() - sender.startTime) * amtPerSec;
         // Set negative deltas to clear deltas applied by the previous call to `_startSending`
-        _setDeltasFromNow(weights, -int128(amtPerWeight), endTime, updates);
+        _setDeltasFromNow(receivers, -1, endTime, updates);
     }
 
     /// @notice Makes the user start sending funds.
     /// It applies effects of the sender on all of its receivers.
     /// It doesn't modify the sender.
     /// @param sender The updated sender
-    /// @param weights The list of the user's receivers and their weights.
+    /// @param receivers The list of the user's receivers.
     /// @param updates The list of stream updates to log
     function _startSending(
         Sender storage sender,
-        ReceiverWeight[] calldata weights,
+        Receiver[] calldata receivers,
         StreamUpdates memory updates
     ) internal {
-        uint32 weightsSum = _weightsSum(weights);
+        uint128 amtPerSec = _totalAmtPerSec(receivers);
```

Comment:
I think it would be cheaper to let the user provide the `_totalAmtPerSec` as a parameter. 

A iteration is happening in the `_setDeltasFromNow`. Here we can calculate the acutal totalAmtPerSec which could be returned. 

If the provided `totalAmtPerSec` doesn't match with the actual one we `revert`.

In that way we force the user to provide the correct one

Reply:
We've discussed it here: https://github.com/radicle-dev/radicle-streaming/pull/44#discussion_r736018052, https://github.com/radicle-dev/radicle-streaming/pull/44#discussion_r736020501, https://github.com/radicle-dev/radicle-streaming/pull/44#discussion_r73602260, https://github.com/radicle-dev/radicle-streaming/pull/41#discussion_r736001457 and we've opened an issue for that: https://github.com/radicle-dev/radicle-streaming/issues/46